core/src/main/java/hudson/FilePath.java

```diff
@@ -2705,18 +2706,17 @@ public boolean isRemote() {
     }
 
     private void writeObject(ObjectOutputStream oos) throws IOException {
-        Channel target = Channel.current();
-
-        if(channel!=null && channel!=target)
+        Channel target = getChannelForSerialization();
+        if(channel!=target) {
```

Comment:
Note that any code which used to serialize a `FilePath` for some other reason (basic saving just `remote`) is going to start failing now. Which is probably a good thing.

(Pipeline `program.dat` uses `FilePathPickle` instead.)

Reply:
Probably I should revert it. I need to browse the code and to remember why I did it for this class.

Reply:
Acc. to e.g. `BuildCommandTest.sync` this is broken.